edgar-doctor/src/components/app/dashboardPages/patients/subViews/PatientsSubViewAppointments.tsx

```diff
@@ -0,0 +1,120 @@
+'use client';
+
+import { useState } from 'react';
+
+import { Button, HStack, VStack, useBreakpointValue } from '@chakra-ui/react';
+
+import paginationHandler from 'utils/navigation/paginationHandler';
+import Pagination from 'components/navigation/Pagination';
+import countMaxNumberPage from 'utils/navigation/countMaxNumberPage';
+
+import { type PatientType } from 'types/app/dashboard/patients/PatientType';
+import { useGetDoctorAppointmentsQuery } from 'services/request/appointments';
+import AppointmentCard from '../../appointments/AppointmentCard';
+
+const PatientsSubViewAppointments = ({
+	selectedPatient,
+}: {
+	selectedPatient: PatientType | undefined;
+}): JSX.Element => {
+	const isMobile = useBreakpointValue({ base: true, smd: false });
+	const [pageIndexUpcoming, setPageIndexUpcoming] = useState(1);
+	const [pageIndexPast, setPageIndexPast] = useState(1);
+	const [appointmentType, setAppointmentType] = useState<'upcoming' | 'past'>('upcoming');
+	const { data: appointments } = useGetDoctorAppointmentsQuery();
+	const patientAppointments = appointments
+		?.filter((appointment) => appointment.patientId === selectedPatient?.id)
+		?.sort((a, b) => new Date(a.startDate).getTime() - new Date(b.startDate).getTime());
+	const upcomingAppointments = patientAppointments?.filter(
+		(patientAppointment) =>
+			patientAppointment.endDate > new Date().getTime() &&
+			!patientAppointment.appointmentStatus.includes('CANCELED'),
+	);
+	const pastAppointments = patientAppointments?.filter(
+		(patientAppointment) =>
+			patientAppointment.endDate <= new Date().getTime() &&
+			!patientAppointment.appointmentStatus.includes('CANCELED'),
+	);
+
+	return (
+		<>
+			{isMobile ? (
+				<VStack w="100%" spacing="8px">
+					<Button
+						w="100%"
+						size="customSm"
+						variant={appointmentType === 'upcoming' ? 'primary' : 'secondary'}
+						onClick={() => setAppointmentType('upcoming')}
+					>
+						Prochain rendez-vous
+					</Button>
+					<Button
+						w="100%"
+						size="customSm"
+						variant={appointmentType === 'past' ? 'primary' : 'secondary'}
+						onClick={() => setAppointmentType('past')}
+					>
+						Rendez-vous passés
+					</Button>
+				</VStack>
+			) : (
+				<HStack w="100%" spacing="8px">
+					<Button
+						w="100%"
+						size="md"
+						variant={appointmentType === 'upcoming' ? 'primary' : 'secondary'}
+						onClick={() => setAppointmentType('upcoming')}
+					>
+						Prochain rendez-vous
+					</Button>
+					<Button
+						w="100%"
+						size="md"
+						variant={appointmentType === 'past' ? 'primary' : 'secondary'}
+						onClick={() => setAppointmentType('past')}
+					>
+						Rendez-vous passés
+					</Button>
+				</HStack>
+			)}
```

Comment:
You can use a `<Stack>` instead of the `<VStack>` and `<HStack>` because you can provide a direction based on the breakpoint, with that you can avoid duplicated code